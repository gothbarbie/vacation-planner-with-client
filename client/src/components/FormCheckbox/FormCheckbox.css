.form-checkbox {
  cursor: pointer;
  margin-top: 1rem;
  padding-left: 1.5rem;
  position: relative;
  user-select: none; }

.form-checkbox__input {
  position: absolute;
  opacity: 0;
  cursor: pointer; }

.form-checkbox__checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 1.5rem;
  width: 1.5rem;
  background-color: white;
  border: 1px solid #80d1ff; }

.form-checkbox__label {
  font-family: 'Avenir Next', sans-serif;
  font-size: 1rem;
  margin-left: .7rem;
  line-height: 1.5rem; }

.form-checkbox:hover .form-checkbox__input ~ .form-checkbox__checkmark {
  background-color: white;
  border: 1px solid #80d1ff; }

.form-checkbox:hover .form-checkbox__checkmark:after {
  display: block; }

.form-checkbox:hover .form-checkbox__checkmark:after {
  left: .45rem;
  top: .15rem;
  width: .4rem;
  height: .8rem;
  border: solid #f2f2f2;
  border-width: 0 3px 3px 0;
  transform: rotate(45deg); }

.form-checkbox .form-checkbox__input:checked ~ .form-checkbox__checkmark {
  background-color: #80d1ff; }

.form-checkbox:hover .form-checkbox__input:checked ~ .form-checkbox__checkmark {
  background-color: white; }

.form-checkbox__checkmark:after {
  content: "";
  position: absolute;
  display: none; }

.form-checkbox .form-checkbox__input:checked ~ .form-checkbox__checkmark:after {
  display: block; }

.form-checkbox .form-checkbox__checkmark:after {
  left: .45rem;
  top: .15rem;
  width: .4rem;
  height: .8rem;
  border: solid white;
  border-width: 0 3px 3px 0;
  transform: rotate(45deg); }
