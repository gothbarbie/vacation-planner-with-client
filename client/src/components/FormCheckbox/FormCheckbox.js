// @flow

import React from 'react'

import './FormCheckbox.css'

type Props = {
  checked: boolean,
  label?: string,
}

const FormCheckbox = ({ checked, label }: Props) => (
  <label className="form-checkbox">
    <input checked={checked} className="form-checkbox__input" type="checkbox" />
    <span className="form-checkbox__checkmark" />
    <label className="form-checkbox__label">{label || 'Label missing'}</label>
  </label>
)

export default FormCheckbox
