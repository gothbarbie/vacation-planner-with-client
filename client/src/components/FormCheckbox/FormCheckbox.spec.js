import React from 'react'
import { shallow } from 'enzyme'
import FormCheckbox from './FormCheckbox'

describe('components/FormCheckbox', () => {
  const formCheckbox = shallow(<FormCheckbox />)

  it('renders', () => {
    expect(formCheckbox).toMatchSnapshot()
  })
})
