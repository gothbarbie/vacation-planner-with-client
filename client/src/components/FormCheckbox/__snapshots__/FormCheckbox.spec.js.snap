// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FormCheckbox renders 1`] = `
<label
  className="form-checkbox"
>
  <input
    className="form-checkbox__input"
    type="checkbox"
  />
  <span
    className="form-checkbox__checkmark"
  />
  <label
    className="form-checkbox__label"
  >
    Label missing
  </label>
</label>
`;
